aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -1698,6 +1717,25 @@ func resourceAviatrixSpokeGatewayUpdate(d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	if d.HasChange("enable_preserve_as_path") {
+		enableBgp := d.Get("enable_bgp").(bool)
+		enableSpokePreserveAsPath := d.Get("enable_preserve_as_path").(bool)
+		if enableSpokePreserveAsPath && !enableBgp {
+			return fmt.Errorf("enable_preserve_as_path is not supported for Non-BGP Spoke during Spoke Gateway update")
+		}
+		if !enableSpokePreserveAsPath {
+			err := client.DisableSpokePreserveAsPath(&goaviatrix.SpokeVpc{GwName: gateway.GwName})
+			if err != nil {
+				return fmt.Errorf("could not disable spoke preserve as path: %v", err)
```

Comment:
This error message should be different from the error message given during create so that it's clear where the error occurred. It should include `during Spoke Gateway update` at the end. Something like: `could not disable Spoke Preserve AS Path during Spoke Gateway update: %v`